src/routes/builder/index.svelte

```diff
@@ -86,6 +90,36 @@
 	$: completions = formContext?.completions;
 	$: computedQuery = formContext?.computedQuery;
 	$: response = formContext?.response;
+	$: {
+		if ($formMachine) {
+			isResponsePending = $formMachine.matches({
+				SelectionComplete: {
+					QueryReady: {
+						Dirty: "Pending"
+					}
+				}
+			});
+			isResponseMatching = $formMachine?.matches({
+				SelectionComplete: {
+					QueryReady: "Matching"
+				}
+			});
+			isResponseError = $formMachine?.matches({
+				SelectionComplete: {
+					QueryReady: {
+						Dirty: 'Error'
+					}
+				}
+			});
+		}
+	}
```

Comment:
I'd append a `derived` store `responseStatus` to `formMachine` context, with 3 booleans `{isPending, isMatching, isError}`, the route should never know the loigic that drives the controller